pretrain/openwebtext.py

```diff
@@ -101,7 +101,7 @@ def main(fabric: L.Fabric, resume: Union[bool, Path]) -> None:
     state = {"model": model, "optimizer": optimizer, "hparams": hparams, "iter_num": 0, "step_count": 0}
 
     if resume is True:
-        resume = sorted(out_dir.glob("*.pth"))[-1]
+        resume = sorted(out_dir.glob("*.pth"), key=lambda p: int(p.name.split("-")[1]))[-1]
```

Comment:
```suggestion
        resume = max(out_dir.glob("*.pth"), key=lambda p: int(p.name.split("-")[1]))
```

Reply:
This fix is incorrect as the sorted list of strings contain both alphabets and numbers, thus taking the max may not take the string with the highest `iter_num`. 

```
from pathlib import Path
out_dir = Path("out") / "openwebtext"
paths = [out_dir / f"iter-{iter_num}-ckpt.pth" for iter_num in range(0, 1100000, 40000)]
print("before fix")
print(sorted(paths))

print("after fix")
sorted_paths = sorted(paths, key=lambda x: int(x.name.split("-")[1]))
print(sorted_paths)
print(max(sorted_paths))
```

returns
```
before fix
[PosixPath('out/openwebtext/iter-0-ckpt.pth'), PosixPath('out/openwebtext/iter-1000000-ckpt.pth'), PosixPath('out/openwebtext/iter-1040000-ckpt.pth'), PosixPath('out/openwebtext/iter-1080000-ckpt.pth'), PosixPath('out/openwebtext/iter-120000-ckpt.pth'), PosixPath('out/openwebtext/iter-160000-ckpt.pth'), PosixPath('out/openwebtext/iter-200000-ckpt.pth'), PosixPath('out/openwebtext/iter-240000-ckpt.pth'), PosixPath('out/openwebtext/iter-280000-ckpt.pth'), PosixPath('out/openwebtext/iter-320000-ckpt.pth'), PosixPath('out/openwebtext/iter-360000-ckpt.pth'), PosixPath('out/openwebtext/iter-40000-ckpt.pth'), PosixPath('out/openwebtext/iter-400000-ckpt.pth'), PosixPath('out/openwebtext/iter-440000-ckpt.pth'), PosixPath('out/openwebtext/iter-480000-ckpt.pth'), PosixPath('out/openwebtext/iter-520000-ckpt.pth'), PosixPath('out/openwebtext/iter-560000-ckpt.pth'), PosixPath('out/openwebtext/iter-600000-ckpt.pth'), PosixPath('out/openwebtext/iter-640000-ckpt.pth'), PosixPath('out/openwebtext/iter-680000-ckpt.pth'), PosixPath('out/openwebtext/iter-720000-ckpt.pth'), PosixPath('out/openwebtext/iter-760000-ckpt.pth'), PosixPath('out/openwebtext/iter-80000-ckpt.pth'), PosixPath('out/openwebtext/iter-800000-ckpt.pth'), PosixPath('out/openwebtext/iter-840000-ckpt.pth'), PosixPath('out/openwebtext/iter-880000-ckpt.pth'), PosixPath('out/openwebtext/iter-920000-ckpt.pth'), PosixPath('out/openwebtext/iter-960000-ckpt.pth')]
after fix
[PosixPath('out/openwebtext/iter-0-ckpt.pth'), PosixPath('out/openwebtext/iter-40000-ckpt.pth'), PosixPath('out/openwebtext/iter-80000-ckpt.pth'), PosixPath('out/openwebtext/iter-120000-ckpt.pth'), PosixPath('out/openwebtext/iter-160000-ckpt.pth'), PosixPath('out/openwebtext/iter-200000-ckpt.pth'), PosixPath('out/openwebtext/iter-240000-ckpt.pth'), PosixPath('out/openwebtext/iter-280000-ckpt.pth'), PosixPath('out/openwebtext/iter-320000-ckpt.pth'), PosixPath('out/openwebtext/iter-360000-ckpt.pth'), PosixPath('out/openwebtext/iter-400000-ckpt.pth'), PosixPath('out/openwebtext/iter-440000-ckpt.pth'), PosixPath('out/openwebtext/iter-480000-ckpt.pth'), PosixPath('out/openwebtext/iter-520000-ckpt.pth'), PosixPath('out/openwebtext/iter-560000-ckpt.pth'), PosixPath('out/openwebtext/iter-600000-ckpt.pth'), PosixPath('out/openwebtext/iter-640000-ckpt.pth'), PosixPath('out/openwebtext/iter-680000-ckpt.pth'), PosixPath('out/openwebtext/iter-720000-ckpt.pth'), PosixPath('out/openwebtext/iter-760000-ckpt.pth'), PosixPath('out/openwebtext/iter-800000-ckpt.pth'), PosixPath('out/openwebtext/iter-840000-ckpt.pth'), PosixPath('out/openwebtext/iter-880000-ckpt.pth'), PosixPath('out/openwebtext/iter-920000-ckpt.pth'), PosixPath('out/openwebtext/iter-960000-ckpt.pth'), PosixPath('out/openwebtext/iter-1000000-ckpt.pth'), PosixPath('out/openwebtext/iter-1040000-ckpt.pth'), PosixPath('out/openwebtext/iter-1080000-ckpt.pth')]
out/openwebtext/iter-960000-ckpt.pth
```

note that `iter-960000` is returned when the highest is actually `iter-1080000`.

Reply:
Hello @larrylawl

Thanks for the answer. I like that you added this code snippet 👍

But there is an error in your example. 
I don't suggest to apply `max` on a sorted list, but rather to do it on an unsorted list (though nothing stops you from doing it with sorted one),  the only thing is that you have to provide `key` value to this function anyway, so it knows what value to take into consideration when deciding what should be the max item.

Reply:
Ahh, sorry I misread your change. Your change looks better - thanks!